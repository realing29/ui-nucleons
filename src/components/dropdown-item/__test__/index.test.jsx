import React from 'react';
import { mount } from 'enzyme';
import { DropdownItem } from '..';

describe('DropdownItem', () => {
  it('should render without props', () => {
    const wrapper = mount(
      <DropdownItem />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render as disabled', () => {
    const wrapper = mount(
      <DropdownItem disabled />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with no hover effect', () => {
    const wrapper = mount(
      <DropdownItem noHover />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle props', () => {
    const wrapper = mount(
      <DropdownItem className='hello' role='test-role' size='l'>
        Hello, world!
      </DropdownItem>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle size prop', () => {
    ['s', 'm', 'l', 'xl'].forEach(size => {
      const wrapper = mount(
        <DropdownItem size={size} />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
