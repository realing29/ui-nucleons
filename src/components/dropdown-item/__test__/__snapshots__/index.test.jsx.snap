// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem should handle props 1`] = `
<DropdownItem
  className="hello"
  role="test-role"
  size="l"
>
  <div
    className="root size-l hello"
    role="test-role"
  >
    Hello, world!
  </div>
</DropdownItem>
`;

exports[`DropdownItem should handle size prop 1`] = `
<DropdownItem
  size="s"
>
  <div
    className="root size-s"
  />
</DropdownItem>
`;

exports[`DropdownItem should handle size prop 2`] = `
<DropdownItem
  size="m"
>
  <div
    className="root size-m"
  />
</DropdownItem>
`;

exports[`DropdownItem should handle size prop 3`] = `
<DropdownItem
  size="l"
>
  <div
    className="root size-l"
  />
</DropdownItem>
`;

exports[`DropdownItem should handle size prop 4`] = `
<DropdownItem
  size="xl"
>
  <div
    className="root size-xl"
  />
</DropdownItem>
`;

exports[`DropdownItem should render without props 1`] = `
<DropdownItem>
  <div
    className="root size-m"
  />
</DropdownItem>
`;
